plugins/flatpak/gs-flatpak.c

```diff
@@ -323,6 +323,22 @@ gs_flatpak_add_flatpak_keyword_cb (XbBuilderFixup *self,
 	return TRUE;
 }
 
+static gboolean
+gs_flatpak_mark_endlessm_apps_popular_cb (XbBuilderFixup *self,
+					   XbBuilderNode *bn,
+					   gpointer user_data,
+					   GError **error)
+{
+	if (g_strcmp0 (xb_builder_node_get_element (bn), "component") == 0) {
+		g_autoptr(XbBuilderNode) id = xb_builder_node_get_child (bn, "id", NULL);
```

Comment:
It’s possible for `id` to be `NULL` if the XML is malformed (a `<component>` doesn’t contain an `<id>`), which will cause a crash.